### ISSUE TITLE
Failed to resolve NFD-master DNS

### ISSUE THREAD

**Issue by hhk7734** [NVIDIA/gpu-operator#320]
_The template below is mostly useful for bug reports and support questions. Feel free to remove anything which doesn't apply to you and add more information where it makes sense._

### 1. Quick Debug Checklist
- [ ] Are you running on an Ubuntu 18.04 node?
- [x] Are you running Kubernetes v1.13+?
- [x] Are you running Docker (>= 18.06) or CRIO (>= 1.13+)?
- ? Do you have `i2c_core` and `ipmi_msghandler` loaded on the nodes?
- [x] Did you apply the CRD (`kubectl describe clusterpolicies --all-namespaces`)

### 1. Issue or feature description

I installed gpu-operator with

```shell
helm upgrade gpu-operator nvidia/gpu-operator \
    --install \
    --version v1.9.1 \
    -n nvidia \
    --create-namespace \
    --set driver.enabled=false \
    --set toolkit.enabled=false
```

`gpu-operator-node-feature-discovery-worker` log

```shell
W0209 15:45:58.067334 1 clientconn.go:1303] grpc: addrConn.createTransport failed to connect to
{gpu-operator-node-feature-discovery-master:8080 <nil> 0 <nil>}. Err :connection error:
desc = "transport: Error while dialing dial tcp 10.96.189.1:8080: connect: connection refused". Reconnecting...
```

I manually changed `--server=gpu-operator-node-feature-discovery-master:8080` to `--server=gpu-operator-node-feature-discovery-master.nvidia:8080` in worker DaemonSet. and then it was solved.

Probably fixing the below part will solve this error, but I am not  familiar with helm charts :(.

https://github.com/NVIDIA/gpu-operator/blob/b9cecb7cacc79cd100d56cc5b7999fe54c063819/deployments/gpu-operator/charts/node-feature-discovery/templates/worker.yaml#L43-L45

**Comment by hhk7734** [comment]
I kept changing the gpu-operator helm value and something went wrong in the process.
I tried to reproduce this. Master and worker are started in the same namespace. The connection is clear.

The IP has already been found in the error message, and this means that the DNS resolver worked normally.

Re-running the Pod while changing the DaemonSet seems to have solved some other problem. But I thought namespace solved it.

Or... while testing for a long time, I encountered various errors such as tcp timeout, connection refused, etc., but I may have written other error logs as I uploaded the issue while organizing it at the end.

Anyway, sorry for the confusion by posting the wrong issue.
